### ISSUE TITLE
CPU Raises 10-20% For Each New Conference

### ISSUE THREAD

**Issue by E2-Kavinkumar** [jitsi/docker-jitsi-meet#1592]
**Description**

Facing High CPU usage issue which results in call drops. Tried using both **self hosted Jitsi server** and **self hosted jitsi docker server.**

Below are some test results recorded with all audio and video enabled.
- Single call with 2 participants - CPU usage - **1%**
- Single call with 3 participants - CPU usage - **20%**
- Two calls with 3 participants in each - CPU usage - **35-40%**

Increasing the **number of calls with 3 participants in each call** results in a very high CPU usage of the JVB process. This results in participants disconnecting from the call.

**Environment details**

- AWS instance with
     - Processor: Intel(R) Xeon(R) Platinum 8259CL CPU @ 2.50GHz
     - Cores: **8**
     - RAM: **32 Gb**
     - OS: Ubuntu- 22.04.2 LTS (Jammy Jellyfish)

Attaching the docker configuration files.
[env.txt](https://github.com/jitsi/docker-jitsi-meet/files/12463157/env.txt)
[docker-compose.txt](https://github.com/jitsi/docker-jitsi-meet/files/12463162/docker-compose.txt)

The similar configuration we used in the self hosted Jitsi server.

Kindly point us in the right direction to resolve this ASAP.
Thanks in advance and let me know if you need any more details.

**Comment by E2-Kavinkumar** [comment]
Adding to that these test results were taken in a fresh AWS instance and also no other process were running at the time of that testing. 

Refer the below screenshot.
![Jitsi_cpu_usage](https://github.com/jitsi/docker-jitsi-meet/assets/65644780/58df1825-3711-4ad0-8245-3841429b76bb)

**Comment by saghul** [comment]
IIRC that's 20% of one core, over the sampling period, so you still have plenty of headroom.

**Comment by E2-Kavinkumar** [comment]
@saghul Yeah, I can see that. That 20% is recorded for a single call with 3 users. The CPU usage is increasing exponentially when I connect more users. Also I have users disconnect issue when the number of users count reaches around 12 to 15. That's the major issue I have. Kindly let me know if you need any more clarifies in it. Waiting for some solutions to resolve this.

**Comment by saghul** [comment]
> Yeah, I can see that. That 20% is recorded for a single call with 3 users. The CPU usage is increasing exponentially when I connect more users.

How many did you connect and how far did it go? Note it may take a bit to normalize.



> Also I have users disconnect issue when the number of users count reaches around 12 to 15.

That sounds like something else. We'd need browser console logs to star diagnosing.

**Comment by damencho** [comment]
Can you check jvb logs for similar logs?

```
2023-08-29 19:22:00.057 INFO: [274] JitsiOpenSslProvider.<clinit>#52: jitsisrtp successfully loaded for OpenSSL 3
2023-08-29 19:22:00.401 INFO: [274] Aes.benchmark#367: AES benchmark (of execution times expressed in nanoseconds): OpenSSL 1268, SunJCE 976, BouncyCastle 10972 for AES/CTR/NoPadding
2023-08-29 19:22:00.402 INFO: [274] Aes.createCipher#433: Will employ AES implemented by SunJCE for AES/CTR/NoPadding.
```

**Comment by E2-Kavinkumar** [comment]
@damencho  Yes I found some. 

![image](https://github.com/jitsi/docker-jitsi-meet/assets/65644780/cb005501-4a5c-4ac1-bbac-fc9ef00b9c37)

**Comment by E2-Kavinkumar** [comment]
Sorry.I searched for "AES benchmark" and "Aes.benchmark". but nothing found. May I know what is it about Aes.benchmark?

**Comment by damencho** [comment]
You can see in the logs I pasted, it compares different implementations that are used for decrypt/encrypt packets, which is the most intensive work done in jvb.
In general there are reports for Openssl 3 being slow, I wonder whether this is it ...

**Comment by E2-Kavinkumar** [comment]
Thanks for the details. Is there any configuration to double check that this is not the root cause or prevent this from happening?

**Comment by damencho** [comment]
I'm not sure there is a problem here. I was checking a bridge on meet.jit.si that was having 110+ local endpoints and one of the threads is getting more than 200% of cpu and the load on the machine is close to 4 same number of cores.

Have you seen what is the load when there are disconnects? Have you tested the limits of the machine, how many participants it can get?

**Comment by E2-Kavinkumar** [comment]
Apologies for the confusion. 
After a whole lot of testing and debugging in network we came to know that this issue is due to a low network bandwidth. It took us almost a while sort it out as we were focusing on the CPU usage of JITSI server. Still we can see the CPU spikes and usage increasing based on number of calls and users but that does not concern us anymore as we do not see any call drops and everything seems to be fine.
Closing this ticket with the above details

Anyway thanks for your response @damencho  and @saghul
